### ISSUE TITLE
Publish on crates.io

### ISSUE THREAD

**Issue by Shock-1** [inferiorhumanorgans/sam3x8e-rs#1]
you can publish the crates using cargo publish. It would give it some visibilty, and even recently someone was searching for sam3x support.

**Comment by inferiorhumanorgans** [comment]
My intent was to flesh out a few more features before pushing to crates.io.  I was hoping to make more progress more quickly but unfortunately I've been sidetracked.

**Comment by Shock-1** [comment]
Well it's pretty complete for a lot of cases, release quick release often as they say.

**Comment by inferiorhumanorgans** [comment]
Apologies for the delay.  If you're still interested:

The reason I didn't publish anything to crates.io is that publishing entails a bit of a commitment and the crate is entirely usable from a git repository.  The API is/was still wildly in flux and I was hoping to wait until I had an MVP before publishing anything.

I've picked this project up again, unfortunately my main use case (the Macchina M2) has been abandoned by the manufacturer.  Right now the MVP is: read/write support for the UART, USART, and UOTGHS (USB) peripherals.  After that the DMAC and CAN peripherals.

I've cleaned up the API a bit and sorted through some of my poor interpretations of the datasheet.  Certainly if you're still interested I'd welcome the feedback on where the API is going and how well things work.

I'm going to archive this repository and close the issue as I've moved development over to Sourcehut to avoid feeding Microsoft's abysmal UX team and AI monster.

https://git.sr.ht/~az1/sam3x-rs/
